File: python/python/Ice/UnknownSlicedValue.py
# Copyright (c) ZeroC, Inc.

import IcePy

from .Value import Value


class UnknownSlicedValue(Value):
    """
    Represents an instance of an unknown class.

    Instances of this class are created by the Ice runtime during unmarshaling; they should not be constructed
    directly. A directly constructed instance has no ``unknownTypeId``.

    Attributes
    ----------
    unknownTypeId : str
        The Slice type ID of the unknown value, or the string form of the compact type ID (for example, ``"1"``)
        when the most-derived slice was marshaled with a compact type ID.
    """

    _ice_type = None  # Will be set after class definition
    unknownTypeId: str

    def ice_id(self) -> str:
        """
        Returns the Slice type ID associated with this instance.

        Returns
        -------
        str
            The Slice type ID of the unknown value, or the string form of the compact type ID (for example,
            ``"1"``) when the most-derived slice was marshaled with a compact type ID.
        """
        return self.unknownTypeId


IcePy._t_UnknownSlicedValue = IcePy.defineValue(
    "::Ice::UnknownSlicedValue", UnknownSlicedValue, -1, (), False, None, ()
)

UnknownSlicedValue._ice_type = IcePy._t_UnknownSlicedValue

__all__ = ["UnknownSlicedValue"]

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/UnknownSlicedValue.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/** Represents an instance of an unknown class. */
public final class UnknownSlicedValue extends Value {
    /**
     * Constructs the placeholder instance.
     *
     * @param unknownTypeId the Slice type ID of the unknown value, or the string form of the
     *     compact type ID (for example, "1") when the most-derived slice was marshaled with a
     *     compact type ID
     */
    public UnknownSlicedValue(String unknownTypeId) {
        _unknownTypeId = unknownTypeId;
    }

    /**
     * Returns the Slice type ID associated with this instance.
     *
     * @return the type ID supplied to the constructor; it's the string form of the compact type ID
     *     (for example, "1") when the most-derived slice was marshaled with a compact type ID
     */
    @Override
    public String ice_id() {
        return _unknownTypeId;
    }

    /** The Slice type ID of the unknown value. */
    private final String _unknownTypeId;

    private static final long serialVersionUID = -3199177633147630863L;
}
